packages/discord.js/src/structures/MentionableSelectMenuInteraction.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+const { Collection } = require('@discordjs/collection');
+const MessageComponentInteraction = require('./MessageComponentInteraction');
+const Events = require('../util/Events');
+
+/**
+ * Represents a {@link ComponentType.MentionableSelect} select menu interaction.
+ * @extends {MessageComponentInteraction}
+ */
+class MentionableSelectMenuInteraction extends MessageComponentInteraction {
+  constructor(client, data) {
+    super(client, data);
+
+    const { members, users, roles } = data.data.resolved || {};
+
+    /**
+     * Collection of the selected users
+     * @type {Collection<Snowflake, User>}
+     */
+    this.users = new Collection();
+
+    /**
+     * Collection of the selected users
+     * @type {Collection<Snowflake, GuildMember|APIGuildMember>}
+     */
+    this.members = new Collection();
+
+    /**
+     * Collection of the selected roles
+     * @type {Collection<Snowflake, Role|APIRole>}
+     */
+    this.roles = new Collection();
+
+    if (members) {
+      for (const [id, member] of Object.entries(members)) {
+        const user = users[id];
+        if (!user) {
+          this.client.emit(
+            Events.Debug,
+            `[MentionableSelectMenuInteraction] Received a member without a user, skipping ${id}`,
+          );
+
+          continue;
+        }
+
+        this.members.set(id, this.guild?.members._add({ user, ...member }) ?? { user, ...member });
+      }
+    }
+
+    if (users) {
+      for (const user of Object.values(users)) {
+        this.users.set(user.id, this.client.users._add(user));
+      }
+    }
```

Comment:
```suggestion
    if (users) {
      for (const [id, user] of Object.entries(users)) {
        this.users.set(id, this.client.users._add(user));
        const member = members?.[id];
        if (member) {
          this.members.set(id, this.guild?.members._add({ user, ...member }) ?? { user, ...member });
        }
      }
    }
```
As an alternative to @vladfrangu's suggestion

Reply:
In practice this does the same thing my current code does, it doesn't actually address Vlad's review.

Reply:
> In practice this does the same thing my current code does

But with 1 loop. 2 times faster.

> It doesn't actually address Vlad's review

No skipping, no the Vlad's review

Reply:
It.. still very much skips though in the impossible edge case that there's a guild member with no associated user, just that now there's no debug log/explicitness to it.

Reply:
Discord gives us a promise that the member will be there with the user if the interaction is from a guild as it gives us 
a promise that by route `PATCH: /channels/{channel.id}/messages/{message.id}` we edit a message, not delete it.
Also that case isn't documented so I don't think we need that debug log.

Reply:
I'm glad you're so optimistic, but regardless, the log was added under the guidance of fellow contribs/core maintainers, so it's already been decided that it's a necessary sanity check.